src/apps/dictionaries/redux/actions.ts

```diff
@@ -391,11 +392,19 @@ export const recursivelyAddConceptsToDictionaryAction = (
     );
 
     const referencesToAdd = await recursivelyFetchToConcepts(
-      sourceUrl,
       concepts.map(concept => concept.id),
-      updateProgress
+      updateProgress,
+      sourceUrl,
+      false
     );
 
+    const groupedConcepts = groupBy(concepts.sourceUrl),
+     const referencesToAdd = Object.entries(groupedConcepts).map(([source, concepts]) => recursivelyFetchToConcepts(
+       concepts.sourceUrl,
```

Comment:
The `sourceUrl` should be the `source` argument. Remember, however, that the `sourceUrl` now needs to be the *last* argument to this, if we're re-arranging the argument order for `recursivelyFetchToConcepts()`